components/shipstation/actions/create-order/create-order.mjs

```diff
@@ -0,0 +1,100 @@
+import shipstation from "../../shipstation.app.mjs";
+import constants from "../common/constants.mjs";
+import utils from "../common/utils.mjs";
+
+export default {
+  key: "shipstation-create-order",
+  name: "Create Order",
+  description: "Create a new order. [See docs here](https://www.shipstation.com/docs/api/orders/create-update-order)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shipstation,
+    orderNumber: {
+      label: "Order Number",
+      description: "A user-defined order number used to identify an order.",
+      type: "string",
+    },
+    orderDate: {
+      label: "Order Date",
+      description: "The date the order was placed. E.g. 2015-06-29T08:46:27.0000000",
+      type: "string",
+    },
+    orderStatus: {
+      label: "Order Status",
+      description: "The order's status.",
+      type: "string",
+      options: constants.ORDER_STATUSES,
+    },
+    billTo: {
+      label: "Bill To",
+      description: "The recipients billing address. Use the [Address](https://www.shipstation.com/docs/api/models/address) model.",
+      type: "string",
+    },
+    shipTo: {
+      label: "Ship To",
+      description: "The recipient's shipping address. Use the [Address](https://www.shipstation.com/docs/api/models/address) model.",
+      type: "string",
+    },
+    customerUsername: {
+      label: "Customer Username",
+      description: "The customer's username. Note: This property needs to be defined in order to generate a customer profile",
+      type: "string",
+      optional: true,
+    },
+    customerEmail: {
+      label: "Customer Email",
+      description: "The customer's email address.",
+      type: "string",
+      optional: true,
+    },
```

Comment:
Could `customerUsername` and/or `customerEmail` use async options in case the user wants to select an existing customer? https://www.shipstation.com/docs/api/customers/list/